### ISSUE TITLE
QCAlgorithm.OnEverythingInitialized or similar

### ISSUE THREAD

**Issue by 4real** [QuantConnect/Lean#1043]
Here's my problem: I need to do some stuff after all the brokerage initialization has taken place, e.g. to synchronize my algorithm with open order list due to restarting it. Brokerage connection/initialization takes place after QCAlgorithm.Initialize, so I can't put my logic there.

I can of course achieve this via various ad-hoc methods (doing something in first OnData call, setting a scheduled event to fire ASAP and then remove it, etc.), but none of them are fully reliable. Plus, while first OnData call would be a fairly clean choice, it means my code won't actually execute until that call (meaning I can't debug it outside market hours).

To summarize, while I can solve this problem presently, it ends up requiring far more thinking than is reasonable for something that should be simple.

Of course, I presume it's very easy to add this (hardest part is coming up with a name that does not confuse the user).
